src/components/FeatureViews/Registration/PaymentReceipt.js

```diff
@@ -1,152 +1,109 @@
-import React, {useCallback, useEffect, useMemo, useState} from "react";
+import React, {useMemo} from "react";
 import {Prompt, useHistory, useLocation, useParams} from "react-router-dom";
 import {useDispatch, useSelector} from "react-redux";
-import {bindActionCreators} from "redux";
 
 import Button from "@material-ui/core/Button";
 import Grid from "@material-ui/core/Grid";
 import Paper from "@material-ui/core/Paper";
 import Typography from "@material-ui/core/Typography";
 
 import * as registrationActions from "actions/registrationActions";
-import * as userActions from "actions/userActions";
-import {isFail, isLoading, isSuccessful, usePrevious} from "actions/hooks";
-import {usePayment, useSubmitRegistration} from "actions/multiCallHooks";
 import BackButton from "components/BackButton";
-import {GET} from "actions/actionTypes";
 import Loading from "components/Loading";
 import {paymentToString, uniques} from "utils";
 import Moment from "react-moment";
+import gql from "graphql-tag";
+import {useQuery} from "@apollo/react-hooks";
+import {bindActionCreators} from "redux";
+import {fullName} from "../../../utils";
 
-const RegistrationReceipt = () => {
+export const GET_PAYMENT = gql`
+	query Payment($paymentId:ID!){
+		payment(paymentId: $paymentId) {
+			id
+			createdAt
+			registrationSet {
+			  id
+			  enrollment {
+				course {
+				  hourlyTuition
+				  title
+				  startDate
+				  endDate
+				  id
+				}
+				id
+				student {
+				  user {
+					lastName
+					firstName
+					id
+				  }
+				}
+			  }
+			  numSessions
+			}
+			total
+			method
+			parent {
+			  user {
+				firstName
+				lastName
+				id
+			  }
+			}
+			priceAdjustment
+			subTotal
+			discountTotal
+		}
+	}
+`;
+
+const PaymentReceipt = ({paymentID}) => {
 	const history = useHistory();
+	const location = useLocation();
+	const params = useParams();
+
+	const {data, loading, error} = useQuery(GET_PAYMENT,
+		{variables: {paymentId: params.paymentID || paymentID}}
```

Comment:
We should probably choose one or the other to use (unless we need both use cases?)

Reply:
yeah, we need both use cases - one for when you go onto a payment page, the other for when you're completing a registration flow.